presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/TableLayoutRewriter.java

```diff
@@ -70,10 +74,12 @@ public PlanNode planTableScan(TableScanNode node, Expression predicate)
                 session, node.getTable(),
                 new Constraint<>(simplifiedConstraint, bindings -> true),
                 Optional.of(ImmutableSet.copyOf(node.getAssignments().values())));
-
         if (layouts.isEmpty()) {
             return new ValuesNode(idAllocator.getNextId(), node.getOutputSymbols(), ImmutableList.of());
         }
+        layouts = layouts.stream()
+                .filter(layout -> layout.hasAllOutputs(node))
+                .collect(toImmutableList());
```

Comment:
I think we should  use `checkState` here and in `AddExchanges` as well.  See line 76 where we ask for a layout with all the node columns, so it should return layouts with all the columns provided.

Reply:
good point.

Reply:
We decided to leave it as it is for now. Because the same code exists in `AddExchanges`, then as a follow up PR we will try to remove it from `PickTableLayout` and from `AddExchanges` at once.